src/components/menu/__internal__/submenu/submenu.component.js

```diff
@@ -132,7 +132,10 @@ const Submenu = React.forwardRef(
 
           // Check that next index contains a MenuItem
           // If not, call handleKeyDown again
-          if (React.Children.toArray(children)[nextIndex].type === MenuItem) {
+          if (
+            React.Children.toArray(children)[nextIndex] &&
```

Comment:
I think we should move `React.Children.toArray` to be on L134 and re-use the value.